### PR TITLE
README: rework description of spec components

### PR DESCRIPTION
**Pull request by erikh**
This replaces the inline links with a bulleted list that clearly states each part of the spec and a little blurb that describes it. I think it's a much easier way to navigate, I hope you agree.

I also made the README link to the spec a lot more prominent.

### CODE REVIEW

**Comment by vbatts** [comment]
@RobDolinMS is this PR what you were referring to?

**Comment by jbouzane** [comment]
I'm concerned about this PR because it presents itself as a formatting change, but it's removing some text from spec.md. Is that intentional, or did it just get lost in the change?

**Comment by erikh** [comment]
What text concerns you specifically? The goal really is to clarify, so if I
made something worse, happy to fix it.

**Comment by stevvooe** [comment]
> I'm concerned about this PR because it presents itself as a formatting change, but it's removing some text from spec.md. Is that intentional, or did it just get lost in the change?

These changes represent a more realistic presentation of the specification.

It looks like there is some language here that was removed for road map items (signatures and naming/federation).

@erikh Would you mind adding those bullets back?

**Comment by erikh** [comment]
I will do it in an hour or so.

-Erik

**Comment by erikh** [comment]
@stevvooe I've made the updates, is this what you're after?

File: spec.md [hunk 1]
```diff
@@ -51,9 +51,12 @@ The [OCI Image Media Types](media-types.md) document is a starting point to unde
 
 The high-level components of the spec include:
 
-* An archival format for container images, consisting of an [image manifest](manifest.md), an [image index](image-index.md) (optional), an [image layout](image-layout.md), a set of [filesystem layers](layer.md), and [image configuration](config.md) (base OCI layer)
-* A [process of referencing container images by a cryptographic hash of their content](descriptor.md) (base OCI layer)
-* A format for [storing CAS blobs and references to them](image-layout.md) (optional OCI layer)
+* [Image Manifest](manifest.md) - the stored document describing an image's properties
```
Comment (jonboulle) [inline spec.md]: Where is the "stored document" stored?

File: spec.md [hunk 2]
```diff
@@ -51,9 +51,12 @@ The [OCI Image Media Types](media-types.md) document is a starting point to unde
...
+* [Image Index](image-index.md) - the stored document describing a collection of images redistributed as one
```
Comment (jonboulle) [inline spec.md]: Is an image index really a collection of images redistributed _as one_? (It sort of implies it's bundled together, is that necessarily the case?)

File: spec.md [hunk 3]
```diff
@@ -51,9 +51,12 @@ The [OCI Image Media Types](media-types.md) document is a starting point to unde
...
+* [Image Layout](image-layout.md) - the filesystem layout representing the contents of an image
+* [Filesystem Layers](layer.md) - the building block of image data
```
Comment (jonboulle) [inline spec.md]: IMHO this description is a bit vague

File: spec.md [hunk 4]
```diff
@@ -51,9 +51,12 @@ The [OCI Image Media Types](media-types.md) document is a starting point to unde
...
+* [Image Configuration](config.md) - the stored document determining layer ordering and configuration of the image suitable for consumption by runtime-spec
+* [Descriptors](descriptor.md) - a concept used to refer to container images by the hash of their content
```
Comment (jonboulle) [inline spec.md]: it's not just container images, it's all components of the spec

File: spec.md [hunk 5]
```diff
@@ -51,9 +51,12 @@ The [OCI Image Media Types](media-types.md) document is a starting point to unde
...
 The high-level components of the spec include:
 
-* An archival format for container images, consisting of an [image manifest](manifest.md), an [image index](image-index.md) (optional), an [image layout](image-layout.md), a set of [filesystem layers](layer.md), and [image configuration](config.md) (base OCI layer)
-* A [process of referencing container images by a cryptographic hash of their content](descriptor.md) (base OCI layer)
-* A format for [storing CAS blobs and references to them](image-layout.md) (optional OCI layer)
+* [Image Manifest](manifest.md) - the stored document describing an image's properties
+* [Image Index](image-index.md) - the stored document describing a collection of images redistributed as one
+* [Image Layout](image-layout.md) - the filesystem layout representing the contents of an image
+* [Filesystem Layers](layer.md) - the building block of image data
+* [Image Configuration](config.md) - the stored document determining layer ordering and configuration of the image suitable for consumption by runtime-spec
```
Comment (jonboulle) [inline spec.md]: Not really suitable for consumption; maybe this would more appropriately link to what comes out of #492

**Comment by erikh** [comment]
Ok. I will pull the information from the other docs; didn't think of that.

**Comment by stevvooe** [comment]
> @stevvooe I've made the updates, is this what you're after?

Yea, the separation looks good.

The descriptions of each component need a little work. I'll comment in line.

File: spec.md (hunk 1 above)
Comment (stevvooe) [inline spec.md]: "Describes the components that make up a container image."

File: spec.md (hunk 3 above)
Comment (stevvooe) [inline spec.md]: "changesets that describe a container's filesystem"

File: spec.md (hunk 4 above)
Comment (stevvooe) [inline spec.md]: "Describes the type, metadata and content address of referenced content."

**Comment by jonboulle** [comment]
Steve's suggestions look good to me

Am 04.04.2017 23:31 schrieb "Stephen Day" <notifications@github.com>:

File: spec.md (hunk 2 above)
Comment (stevvooe) [inline spec.md]: More or less. Effectively, they reference one or more manifests, annotating the underlying content with platform information "pulled up" config. In practice, we could compare this to a "fat binary", although the content isn't embedded.

"An index of annotated image manifests distributed as a complete unit."

Not great, but that might help to better describe this.

**Comment by erikh** [comment]
Thanks for the guidance, should have these incorporated by EOD. I know the schedule is tight.

File: spec.md (hunk 5 above)
Comment (erikh) [inline spec.md]: what do we do for now though?

**Comment by erikh** [comment]
PTAL

**Comment by vbatts** [comment]
LGTM

**Comment by stevvooe** [comment]
LGTM

**Pull request merged**